atomic_reactor/util.py

```diff
@@ -1520,30 +1523,41 @@ def get_log_files(self, osbs, pipeline_run_name):
         """
 
         logs = None
-        output = []
+        outputs = []
 
         # Collect logs from server
         try:
             logs = osbs.get_build_logs(pipeline_run_name)
         except OsbsException as ex:
             self.log.error("unable to get build logs: %s", ex)
-            return output
+            return outputs
 
-        # OSBS2 TBD logs for specific arches for binary builds
         filename = 'osbs-build'
-        logfile = NamedTemporaryFile(prefix=f"{pipeline_run_name}-{filename}",
-                                     suffix=".log", mode='wb')
+        logfiles = {'noarch': NamedTemporaryFile(prefix=f'{pipeline_run_name}-{filename}-noarch-',
+                                                 suffix='.log', mode='wb')}
 
         # Correct log order depends on dict iteration order
         # and on osbs.get_build_logs returning correctly ordered dict
-        for containers in logs.values():
-            for log_message in containers.values():
-                logfile.write((log_message + '\n').encode('utf-8'))
-        logfile.flush()
-        metadata = self.get_log_metadata(logfile.name, f"{filename}.log")
-        output.append(Output(file=logfile, metadata=metadata))
-
-        return output
+        for task_run_name, _ in logs.items():
+            log_platform = 'noarch'
+            for platform in self.platforms:
+                if platform in task_run_name:
+                    if platform not in logfiles:
+                        logfiles[platform] = NamedTemporaryFile(prefix=f'{pipeline_run_name}-'
+                                                                       f'{filename}-{platform}-',
+                                                                suffix='.log', mode='wb')
+                    log_platform = platform
+                    break
+            for log_message in logs[task_run_name].values():
+                logfiles[log_platform].write((log_message + '\n').encode('utf-8'))
+            logfiles[log_platform].flush()
+
+        for platform, _ in logfiles.items():
+            metadata = self.get_log_metadata(logfiles[platform].name,
+                                             os.path.basename(f'{filename}-{platform}.log'))
```

Comment:
Does this change the name of the log file that will be uploaded to Koji? I think we should keep those the way they are currently: "osbs-build.log" for the main one and "{platform}.log" for the others.

But I have to admit, I do not understand which part of OSBS uploads logs where :(

Reply:
yeah here you should include platform just for regular arches, and keep osbs-build.log for noarch

Reply:
updated names accordingly. Please note I've left temp file names as it is since they have 8 characters of random letter added to the name and I am fairly sure the name that is set in brew is the filename in metadata which I have set here.